templates/glossary.md

```diff
@@ -134,9 +137,27 @@ Examples of such commands are the `def`, `theorem`, `constant` or `example` comm
 
 Further detail can be found in the [Lean documentation](https://leanprover.github.io/lean4/doc/declarations.html#basic-declarations).
 
+### dependent type theory
+
+A [type theory](#type-theory) in which one additionally may have types which depend on a parameter, such as the type of "days of a calendar month", whose specific values depend on the particular month, given that different months have different numbers of days.
```

Comment:
Do you find this example clearer than `vector A n`, the type of `n`-tuples of elements of `A`? I'm not reading it as a novice but I feel like I stumbled over the days example more than I would have.

Reply:
I think for me it clicked probably only after seeing a few different examples, but I found days of the month *slightly* and only very slightly easier to understand.

I think the reason was purely that for `vector A n` you have to stress that the length is part of the type, which is both a bit longer to writem as well as possible to miss the importance of if you're of certain background. In my head I mapped `vector A n` to Python list (somehow glossing over the length-part-of-type bit being important), and said "yeah what's the problem?", whereas with days of the month it was clearer that "oh yeah that'd be annoying".

But it's a tricky entry for novices, who knows if I even fully understand it myself yet...